src/AI/agents/agents/services/llm/intent_parser.py

```
@@ -155,10 +155,41 @@ def _validate_goal_safety_quick(goal: str) -> tuple[bool, Optional[str]]:
+    if len(kept) != len(candidates):
+        log.info(f"Dropped {len(candidates) - len(kept)} suggestion(s) the gate would reject")
+    return kept
+
+
+async def _would_be_accepted(goal: str) -> bool:
+    is_safe, _ = _validate_goal_safety_quick(goal)
+    if not is_safe:
+        return False
+    try:
+        parsed = await parse_intent_async(goal)
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)